cupy/cuda/cupy_thrust.cu

```diff
@@ -129,3 +132,40 @@ template void cupy::thrust::_argsort<cpy_long>(size_t *, void *, size_t);
 template void cupy::thrust::_argsort<cpy_ulong>(size_t *, void *, size_t);
 template void cupy::thrust::_argsort<cpy_float>(size_t *, void *, size_t);
 template void cupy::thrust::_argsort<cpy_double>(size_t *, void *, size_t);
+
+
+/*
+ * shuffle
+ */
+
+struct Shuffle
+{
+    uint32_t seed;
+    Shuffle(uint32_t s) { seed = s; }
+    __device__ uint32_t operator()(uint32_t kn) const
+    {
+        minstd_rand randEngine(seed);
```

Comment:
I'm not familiar with behavior of this object in CUDA environment, but is it efficient? I feel it is more efficient to make a random values with curand.